helix/package-principles.rst

```diff
@@ -0,0 +1,137 @@
+owned by the Project layer (see `Template types`_).
+
+.. _Template types: principles/templates/template-types.html
+
+.. note::
+ 
+    **The Common Closure Principle (CCP)**
+
+    The classes in a component should be closed together against the same 
+    kinds of changes. A change that affects a component affects all the 
+    classes in that component and no other components.
+
+This principle seeks to limit the impact of changes to one or a small set of 
+modules, so that changes do not ripple across all the components of a system. 
+This principle motivates the Helix practice of grouping different types of elements
+(templates, renderings, views, controllers) that collaborate to produce a feature.
+In this way, a change to that feature will land entirely within the scope of one
+module. For example, imagine adding a new behavior to a navigation component, 
+with a new field added to an interface template, new code added to a model repository,
+and new markup added to CSHTML. Although these are different technologies, they are 
+bound in one Helix module, and thus the impact of the modification is limited to 
+that module. 
+
+Principles of Component Coupling
+--------------------------------
+The last three principles define how components should interact with one another.
+
+.. note::
+
+  **The Acyclic Dependencies Principle (ADP)**
+  
+  No cycles will be accepted in the dependency graph.
+
+This principle guards against circular references and dependencies, so that the direction of 
+dependencies flow in one direction.  The reason for this precaution is that if a circular 
+dependency is allowed to stay, then a change to the offending component will mean that every other 
+component will need to be validated,  undoing the benefits of modularization.  
+It is important to note that these cycles can arise in 
+numerous ways: project references, which Visual Studio will guard against, but also through 
+more subtle means, such as type references in content or markup. This principle inspires Helix's attention
+to managing relationships through layers, and for taking proactive steps to guard against implicit
+dependencies between modules. For example, Helix requires that fields always be addressed by Field ID 
+rather than field name, as by using field names a module takes on an implicit dependencies on 
+every other module in the system not to have overlapping field names. (See `References from code`_)
+
+.. _References from Code: principles/templates/references.html
+
+.. note::
+
+  **The Stable Dependencies Principle (SDP)**
+  
+  Code should depend in the direction of stability
+
+Components that change rarely should not depend on modules that change often, as this reduces the benefit of 
+modularizing stable code. Martin proposes a numerical formula for assessing module stability in terms of class
+dependencies. In Helix, this mechanism is replaced by two guard rails: all visible features should live in "Feature"-layer 
```

Comment:
Is the word *visible* problematic here as a Feature module could be an integration or other non-visible function?

Reply:
Sure. I think the point that the feature is user-facing is made in the following discussion.

```suggestion
dependencies. In Helix, this mechanism is replaced by two guard rails: all features should live in "Feature"-layer 
```